server/src/main/java/org/opensearch/tasks/consumer/TopNSearchTasksLogger.java

```diff
@@ -36,30 +38,37 @@ public class TopNSearchTasksLogger implements Consumer<Task> {
     private static final Logger SEARCH_TASK_DETAILS_LOGGER = LogManager.getLogger(TASK_DETAILS_LOG_PREFIX + ".search");
 
     // number of memory expensive search tasks that are logged
-    private static final Setting<Integer> LOG_TOP_QUERIES_SIZE_SETTING = Setting.intSetting(
+    public static final Setting<Integer> LOG_TOP_QUERIES_SIZE_SETTING = Setting.intSetting(
         LOG_TOP_QUERIES_SIZE,
         10,
+        1,
+        25,
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
 
     // frequency in which memory expensive search tasks are logged
-    private static final Setting<TimeValue> LOG_TOP_QUERIES_FREQUENCY_SETTING = Setting.timeSetting(
+    public static final Setting<TimeValue> LOG_TOP_QUERIES_FREQUENCY_SETTING = Setting.timeSetting(
         LOG_TOP_QUERIES_FREQUENCY,
         TimeValue.timeValueSeconds(60L),
+        TimeValue.timeValueSeconds(60L),
         Setting.Property.Dynamic,
         Setting.Property.NodeScope
     );
 
-    private final int topQueriesSize;
-    private final long topQueriesLogFrequencyInNanos;
-    private final Queue<Tuple<Long, SearchShardTask>> topQueries;
+    private int topQueriesSize;
+    private long topQueriesLogFrequencyInNanos;
+    private Queue<Tuple<Long, SearchShardTask>> topQueries;
     private long lastReportedTimeInNanos = System.nanoTime();
 
-    public TopNSearchTasksLogger(Settings settings) {
+    public TopNSearchTasksLogger(Settings settings, @Nullable ClusterSettings clusterSettings) {
```

Comment:
Cluster settings should always be non-null. Let's refactor unit-tests as needed.